### ISSUE TITLE
Document recursive modules

### ISSUE THREAD

**Issue by earksiinni** [reason-association/rescript-lang.org#142]
The following syntax is undocumented:

```
module rec MyModule: MyModuleType = {
  ...
}
```

**Comment by ryyppy** [comment]
Yes this should be documented. Open for contributions in the [Module](https://rescript-lang.org/docs/manual/latest/module) section

**Comment by fhammerschmidt** [comment]
One use-case that comes to my mind are recursive React components, for instance for a tree of directories and files.

Example: https://github.com/fhammerschmidt/recursive-folders/blob/main/src/Folder.res#L51

Sorry for answering so late, I guess this one fell under the radar. If you were still willing to submit a PR I would very much appreciate it.

**Comment by Zachiah** [comment]
Sorry didn't see this till now. (You know how github notifications are lol.) I'll try to get to this in the next few days or so.
